src/services/NewsletterSubscriptionService.ts

```diff
@@ -16,17 +20,20 @@ export default new class NewsletterService {
       return await mailing.save()   
     }
     catch (err) {
-      throw new Error(err)
+      throw new Error
     }
   }
 
-  async find () {
-    try {
-      const mailing = Mailing.find()
-      return mailing
+  async find (limit?: number, sortBy?: string, order?: number, startIndex?: number) {
+    try { 
+      return await Mailing.find()
+      .limit(limit)
+      .skip(startIndex)
+      .sort({[sortBy]: order})
```

Comment:
`{[sortBy]: order}` me parece uma sintaxe bem estranha.... Tem certeza que precisa desse `[]` em volta do `sortBy`?

Reply:
Desculpe a inatividade por conta do ano novo, voltei agora com força total!

Nesse caso estou passando o sortBy como um campo de um objeto, esse obj. entra como parâmetro para ordenação na busca no banco de dados, já o order é um valor 1 ou -1 (ascendente ou descendente). Assim quando passamos "email" nesse campo, a ordenação vem de 0 - 9 - A - Z, filtrando pelos emails do banco. Se quisermos porém ordenar por data de criação, no mesmo campo é só passar "createdAt".

Dessa forma o resultado vem de 0 a Z:
/newsletter?page=1&limit=10&sortby=email&order=1 

Dessa forma vem os e-mails criados mais recentemente:
/newsletter?page=1&limit=10&sortby=createdAt&order=-1 

Sobre a sintaxe, essa foi a forma que eu encontrei pra passar uma variável como campo de um objeto, de outra forma simplesmente não ia, mas assim funciona mesmo!

Reply:
Se funciona, funciona 👍